crates/holochain/src/conductor/state.rs

```diff
@@ -62,6 +63,7 @@ pub struct ConductorState {
     pub(crate) conductor_services: ConductorServicesState,
 
     /// List of interfaces any UI can use to access zome functions.
+    #[serde_as(as = "Vec<(_, _)>")]
```

Comment:
Serialize app interfaces as a list instead of a map. Serializing a map would fail with `serde_json` because a json key must be a string or number, and `AppInterfaceId` is a struct with multiple fields.

Reply:
Looks good to me.

`AppInterfaceId` looks weird alright. Maybe it should just be a port. At first sight I don't know why it would need another unique identifier, when ports can only be bound once.

Reply:
Yeah... the docs say that a "0" port means the conductor will pick any open port. So the other string provides a unique identifier when you have multiple "0" ports.

There's something funky in the order of operations here -- why should the conductor need a unique identifier for an app interface that doesn't exist yet? I'm guessing something legacy that wasn't removed fully.

Anyway not to distract right now, I can make an issue for it.